test_db.py

```diff
@@ -0,0 +1,231 @@
+
+        self.con.rollback()
+
+
+# --------------------  SEED FUNCTIONS --------------------
+
+
+class test_seed_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.seed_name = "test"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_get_all_seeds_names(self):
+        """
+        This test checks if the get_all_seeds_names function returns a list of seeds
+        with at least the one seed we added
+        """
+        seed.new_seed(self.cursor, self.seed_name)
+        seeds = seed.get_all_seeds_names(self.cursor)
+
+        self.assertNotEqual(len(seeds), 0)
+        self.assertIn((self.seed_name,), seeds)
+
+        self.con.rollback()
+
+    def test_get_seed_id(self):
+        """
+        This test checks if the get_seed_id function returns the correct UUID
+        """
+        seed_uuid = seed.new_seed(self.cursor, self.seed_name)
+        fetch_id = seed.get_seed_id(self.cursor, self.seed_name)
+
+        self.assertEqual(seed_uuid, fetch_id)
+
+        self.con.rollback()
+
+    def test_new_seed(self):
+        """
+        This test checks if the new_seed function returns a valid UUID
+        """
+        seed_id = seed.new_seed(self.cursor, self.seed_name)
+
+        self.assertTrue(is_valid_uuid(seed_id))
+
+        self.con.rollback()
+
+
+# --------------------  PICTURE FUNCTIONS --------------------
+class test_pictures_functions(unittest.TestCase):
+    def setUp(self):
+        # prepare the connection and cursor
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+
+        # prepare the seed
+        self.seed_name = "test seed"
+        self.seed_id = seed.new_seed(self.cursor, self.seed_name)
+
+        # prepare the user
+        user.register_user(self.cursor, "test@email")
+        self.user_id = user.get_user_id(self.cursor, "test@email")
+
+        # prepare the picture_set and picture
+        self.image = Image.new("RGB", (1980, 1080), "blue")
+        self.image_byte_array = io.BytesIO()
+        self.image.save(self.image_byte_array, format="TIFF")
+        self.pic_encoded = base64.b64encode(self.image_byte_array.getvalue()).decode(
+            "utf8"
+        )
+        self.picture_set = picture_set_data.build_picture_set(self.user_id, 1)
+        self.picture = picture_data.build_picture(
+            self.pic_encoded, self.seed_name, 1, 1.0
+        )
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_new_picture_set(self):
+        """
+        This test checks if the new_picture_set function returns a valid UUID
+        """
+        picture_set_id = picture.new_picture_set(
+            self.cursor, self.picture_set, self.user_id
+        )
+
+        self.assertTrue(
+            is_valid_uuid(picture_set_id), "The picture_set_id is not a valid UUID"
+        )
+
+        self.con.rollback()
```

Comment:
if rollback applies to every test why not have it in teardown?

Reply:
I've moved it to teardown.